runtime/tests/checker/events_test.go

```diff
@@ -1362,4 +1362,33 @@ func TestCheckDefaultEventParamChecking(t *testing.T) {
 		require.IsType(t, &sema.DefaultDestroyInvalidArgumentError{}, errs[0])
 		require.Equal(t, errs[0].(*sema.DefaultDestroyInvalidArgumentError).Kind, sema.ReferenceTypedMemberAccess)
 	})
+
+	t.Run("base outside attachment", func(t *testing.T) {
```

Comment:
Is there also a test case for base inside attachment?

Reply:
Yeah there's an interpreter test for when attachments shadow an external `base` variable